integration_tests/integration/searchForABooking.cy.ts

```diff
@@ -0,0 +1,140 @@
+
+    const childDob = format(sub(today, { years: 5 }), shortDateFormat)
+    const contacts = [
+      TestData.contact({ personId: 1234 }),
+      TestData.contact({
+        personId: 4322,
+        firstName: 'Bob',
+        dateOfBirth: childDob,
+        relationshipCode: 'SON',
+        relationshipDescription: 'Son',
+      }),
+    ]
+
+    homePage.changeAVisitTile().click()
+
+    const searchForBookingByReferencePage = Page.verifyOnPage(SearchForBookingByReferencePage)
+
+    searchForBookingByReferencePage.enterVisitReference('ab-cd-ef-gh')
+
+    cy.task('stubPrisonerById', prisoner)
+    cy.task('stubVisit', visit)
+
+    searchForBookingByReferencePage.continueButton().click()
+
+    searchForBookingByReferencePage.visitReference().contains('ab-cd-ef-gh')
+    searchForBookingByReferencePage.prisonerName().contains(prisonerDisplayName)
+    searchForBookingByReferencePage.prisonerNumber().contains(offenderNo)
+    searchForBookingByReferencePage.visitStatus().contains('Booked')
+
+    cy.task('stubPrisonerSocialContacts', { offenderNo, contacts })
+    cy.task('stubAvailableSupport')
+
+    searchForBookingByReferencePage.visitReferenceLink().click()
+
+    const visitDetailsPage = Page.verifyOnPage(VisitDetailsPage)
+
+    visitDetailsPage.visitReference().contains('ab-cd-ef-gh')
+    visitDetailsPage.prisonerName().contains('Smith, John')
+  })
+
+  it('Should search via prisonerId, than navigate to the summary page', () => {
+    const homePage = Page.verifyOnPage(HomePage)
+    const today = new Date()
+    const prisoner = TestData.prisoner()
+    const { prisonerNumber: offenderNo } = prisoner
+    const prisonerDisplayName = 'Smith, John'
+
+    const upcomingVisit = TestData.visit({
+      reference: 'bc-de-fg-hi',
+      startTimestamp: format(addDays(today, 7), `${shortDateFormat}'T'13:30:00`),
+      endTimestamp: format(addDays(today, 7), `${shortDateFormat}'T'14:30:00`),
+    })
+
+    const childDob = format(sub(today, { years: 5 }), shortDateFormat)
+    const contacts = [
+      TestData.contact({ personId: 1234 }),
+      TestData.contact({
+        personId: 4322,
+        firstName: 'Bob',
+        dateOfBirth: childDob,
+        relationshipCode: 'SON',
+        relationshipDescription: 'Son',
+      }),
+    ]
+
+    homePage.changeAVisitTile().click()
+
+    const searchForBookingByReferencePage = Page.verifyOnPage(SearchForBookingByReferencePage)
+
+    searchForBookingByReferencePage.searchByPrisonerLink().click()
+
+    const searchForBookingByPrisonerPage = Page.verifyOnPage(SearchForBookingByPrisonerPage)
+
+    searchForBookingByPrisonerPage.enterVisitReference(offenderNo)
+
+    cy.task('stubPrisoners', {
+      term: offenderNo,
+      results: {
+        totalElements: 1,
+        totalPages: 1,
+        content: [prisoner],
+      },
+    })
+
+    searchForBookingByPrisonerPage.continueButton().click()
+
+    searchForBookingByPrisonerPage.resultRow().contains(prisonerDisplayName)
```

Comment:
Again, after the `.click()` we're now on a different page, e.g. perhaps something like `searchForBookingByPrisonerResultsPage`.